src/foam/parse/PropertyParser.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.parse;
+
+import java.util.List;
+
+import foam.core.PropertyInfo;
+import foam.lib.json.KeyParser;
+import foam.lib.json.Whitespace;
+import foam.lib.parse.Alt;
+import foam.lib.parse.Literal;
+import foam.lib.parse.PStream;
+import foam.lib.parse.ParserContext;
+import foam.lib.parse.ProxyParser;
+import foam.lib.parse.Seq3;
+import foam.mlang.MLang;
+import foam.mlang.predicate.Predicate;
+
+public class PropertyParser extends ProxyParser {

```

Comment:
I think this would be better named PropertyPredicateParser or PropertyConditionParser as its not parsing a Property value like the foam/lib/json/PropertyParser is, but rather it's parsing a property predicate/conditional.